### PR TITLE
F#src 7087 fix lint

### PR DESCRIPTION
**Pull request by carebare47**
## Proposed changes

Fix new lint errors

## Checklist

Before posting a PR ensure that from each of the below categories **AT LEAST ONE BOX HAS BEEN CHECKED**. If more than one category is applicable then more can be checked. Also ensure that the proposed changes have been filled out with relevant information for reviewers.

## Tests

- [ ] No tests required to be added. (For small changes that will be tested by CI/CD infrastructure).
- [ ] Added/Modified automated and PhantomHand CI tests (if a new class is added (Python or C++), the interface of that class must be unit tested).
- [ ] Manually tested in simulation (if simulation specific or no hardware required to test the functionality). 
- [ ] Manually tested on hardware (if hardware specific or related).

## Documentation

- [x] No documentation required to be added.
- [ ] Added documentation (For any new feature, explain what it does and how to use it. Write the documentation in a relevant space, e.g. Github, Confluence, etc).
- [ ] Updated documentation (For changes to pre-existing features mentioned in the documentation).

### CODE REVIEW

File: sr_utilities/scripts/interactive_marker.py [hunk 1]
```diff
@@ -91,7 +91,7 @@ def processFeedback(self, feedback):
         selected_name = feedback.marker_name
 
         # we loop through all our interactive markers.
-        for index, name in enumerate(self.object_controls):
+        for name in self.object_controls:  # pylint: disable=C0206
```
Comment (maxzieba) [inline sr_utilities/scripts/interactive_marker.py]: instead disabling lints,  self.object_controls.keys() ?

File: sr_utilities/nodes/temperature_monitor.py [hunk 2]
```diff
@@ -34,11 +33,11 @@
...
 
 
-class Joint(object):
+class Joint:
     def __init__(self, screen, joint_name, x, y):
         self.screen = screen
-        self.x = x
-        self.y = y
+        self.x = x  # pylint: disable=C0103
+        self.y = y  # pylint: disable=C0103
```
Comment (maxzieba) [inline sr_utilities/nodes/temperature_monitor.py]: I guess the disabling can be removed with this to be merged https://github.com/shadow-robot/sr-build-tools/pull/662?

File: sr_utilities/scripts/interactive_marker.py [hunk 3]
```diff
@@ -130,5 +130,6 @@ def processFeedback(self, feedback):
 if __name__ == "__main__":
     rospy.init_node("simple_marker")
 
-    int_mark = InteractiveConnectorSelector(["srh/position/palm"], None)
+    # Error here, file probably very depricated
+    int_mark = InteractiveConnectorSelector(["srh/position/palm"], None)  # pylint: disable=E1120
```
Comment (maxzieba) [inline sr_utilities/scripts/interactive_marker.py]: The constructor is def __init__(self, object_names, callback_fct, interactive_server), so 3 args are needed? 
There's a comment above main about it being used for debug, so maybe the main can be removed?

File: sr_utilities/scripts/sr_utilities/controller_spawner.py [hunk 4]
```diff
@@ -93,21 +99,21 @@ def resolve_path(self, path, joint_name=None, local_path=None):
...
         if path.startswith('/'):
             return path
-        else:
-            if local_path is None:
-                return path
-            else:
-                return "{}/{}".format(local_path, path)
+        if local_path is None:
+            return path
+        return "{}/{}".format(local_path, path)
```
Comment (maxzieba) [inline sr_utilities/scripts/sr_utilities/controller_spawner.py]: use f-string

File: sr_utilities/scripts/interactive_marker.py (hunk 3 above)
Comment (carebare47) [reply sr_utilities/scripts/interactive_marker.py]: I'm not sure if anyone is ever gonna try and use this file again, but if they do then maybe `["srh/position/palm"]` as the `object_names` would be a useful hint?

File: sr_utilities/scripts/interactive_marker.py (hunk 1 above)
Comment (carebare47) [reply sr_utilities/scripts/interactive_marker.py]: For me, `for name in self.object_controls` was nicer than `for name in self.object_controls.keys()`. Really we should operate on `for name, object_control in self.object_controls.items()`, but not being able to test this file I thought it better to leave the code alone if possible

File: sr_utilities/nodes/temperature_monitor.py (hunk 2 above)
Comment (carebare47) [reply sr_utilities/nodes/temperature_monitor.py]: yes it can

File: sr_utilities/scripts/sr_utilities/controller_spawner.py (hunk 4 above)
Comment (carebare47) [reply sr_utilities/scripts/sr_utilities/controller_spawner.py]: https://github.com/shadow-robot/sr_core/pull/219/files/c5a8d734f64cab339af59e053e0a30132987ae4b..4862f6b55f76f11a80c6d47a16671c8494115485

File: sr_utilities/nodes/temperature_monitor.py (hunk 2 above)
Comment (carebare47) [reply sr_utilities/nodes/temperature_monitor.py]: it's merged now so removed this disable

File: sr_utilities/scripts/interactive_marker.py (hunk 3 above)
Comment (maxzieba) [reply sr_utilities/scripts/interactive_marker.py]: https://github.com/shadow-robot/sr-demo/blob/c9cb278fdd79fa16899af005ecd0784e3398afc5/sr_unplug_connector/src/unplug_connector_state_machine.py#L123 example here. I'd remove the main as there's no point in it

File: sr_utilities/scripts/interactive_marker.py (hunk 1 above)
Comment (maxzieba) [reply sr_utilities/scripts/interactive_marker.py]: let's wait for someone else to have a look here and come up with some other ideas? :)

**Review by maxzieba — APPROVED** [review]

File: sr_hand/scripts/sr_hand/shadowhand_commander.py [hunk 5]
```diff
@@ -15,43 +15,41 @@
...
-import rospy
 import threading
+import rospy
 
-from sr_robot_msgs.msg import sendupdate, joint
 from sr_hand.shadowhand_ros import ShadowHand_ROS
 from sr_hand.grasps_interpoler import GraspInterpoler
 from sr_hand.Grasp import Grasp
 
-"""
```
Comment (ethanfowler) [inline sr_hand/scripts/sr_hand/shadowhand_commander.py:26]: Should this not stay a docstring? I don't think comments like this are available to e.g. intellisense

File: sr_hand/scripts/sr_hand/shadowhand_commander.py [hunk 6]
```diff
@@ -106,7 +104,7 @@ def move_hand(self, command):
             del joints['interpolation_time']
 
         if interpolation_time == 0.0:
-            self.mutex.acquire()
+            self.mutex.acquire()  # pylint: disable=R1732
```
Comment (ethanfowler) [inline sr_hand/scripts/sr_hand/shadowhand_commander.py]: Why not `with`?

File: sr_hand/scripts/sr_hand/shadowhand_commander.py [hunk 7]
```diff
@@ -127,7 +125,7 @@ def _move_hand_interpolation(self, joints, interpolation_time=1.0):
         rospy.logdebug("call interpolation")
         thread_name = threading.current_thread().name
         try:
-            self.mutex.acquire()
+            self.mutex.acquire()  # pylint: disable=R1732
```
Comment (ethanfowler) [inline sr_hand/scripts/sr_hand/shadowhand_commander.py]: Why not `with`? Especially within a `try`. Mutex might not get released.

File: sr_hand/scripts/sr_hand/shadowhand_commander.py [hunk 8]
```diff
@@ -141,18 +139,18 @@ def _move_hand_interpolation(self, joints, interpolation_time=1.0):
             interpolator = GraspInterpoler(current_grasp, target_grasp)
             self.grasp_interpolators[thread_name] = interpolator
 
-            r = rospy.Rate(1.0 / self.hand_interpolation_period)
+            rate = rospy.Rate(1.0 / self.hand_interpolation_period)
 
             for interpolation in range(1, int(interpolation_time / self.hand_interpolation_period) + 1):
-                self.mutex.acquire()
+                self.mutex.acquire()  # pylint: disable=R1732
```
Comment (ethanfowler) [inline sr_hand/scripts/sr_hand/shadowhand_commander.py]: Why not `with`?

File: sr_hand/scripts/sr_hand/shadowhand_commander.py [hunk 9]
```diff
@@ -141,18 +139,18 @@ def _move_hand_interpolation(self, joints, interpolation_time=1.0):
...
                 targets_to_send = interpolator.interpolate(
                     100.0 * interpolation * self.hand_interpolation_period / interpolation_time)
                 self.mutex.release()
                 self.hand.sendupdate_from_dict(targets_to_send)
                 # rospy.loginfo("sent cmd n: %s" %(str(interpolation), ))
-                r.sleep()
+                rate.sleep()
         finally:
-            self.mutex.acquire()
+            self.mutex.acquire()  # pylint: disable=R1732
```
Comment (ethanfowler) [inline sr_hand/scripts/sr_hand/shadowhand_commander.py]: Why not `with`?

File: sr_tactile_sensors/scripts/sr_tactile_sensors/virtual_ShadowPST_publisher.py [hunk 10]
```diff
@@ -1,6 +1,7 @@
 #!/usr/bin/env python3
+# pylint: disable=C0103
```
Comment (ethanfowler) [inline sr_tactile_sensors/scripts/sr_tactile_sensors/virtual_ShadowPST_publisher.py]: Why? Difficult to rename?

File: sr_utilities/scripts/interactive_marker.py (hunk 1 above)
Comment (ethanfowler) [reply sr_utilities/scripts/interactive_marker.py]: I think `.keys()` is the generally-accepted solution. It's longer but helps readability for me, makes it clear its a dictionary-like we're dealing with.

File: sr_utilities/scripts/interactive_marker.py (hunk 3 above)
Comment (ethanfowler) [reply sr_utilities/scripts/interactive_marker.py]: Agreed, my preferred solution is to remove `main` and this file from installed python executables.

File: sr_utilities/scripts/sr_utilities/controller_spawner.py [hunk 11]
```diff
@@ -17,44 +17,50 @@
...
                    "rh_MFJ1", "rh_MFJ2", "rh_MFJ3", "rh_MFJ4", "rh_RFJ1", "rh_RFJ2", "rh_RFJ3", "rh_RFJ4", "rh_THJ1",
                    "rh_THJ2", "rh_THJ3", "rh_THJ4", "rh_THJ5", "rh_WRJ1", "rh_WRJ2",
                    "lh_FFJ1", "lh_FFJ2", "lh_FFJ3", "lh_FFJ4", "lh_LFJ1", "lh_LFJ2", "lh_LFJ3", "lh_LFJ4", "lh_LFJ5",
                    "lh_MFJ1", "lh_MFJ2", "lh_MFJ3", "lh_MFJ4", "lh_RFJ1", "lh_RFJ2", "lh_RFJ3", "lh_RFJ4", "lh_THJ1",
                    "lh_THJ2", "lh_THJ3", "lh_THJ4", "lh_THJ5", "lh_WRJ1", "lh_WRJ2"]
 
-    def __init__(self, config_file_path, service_timeout, excluded_joints=[]):
-        self._config_file_path = config_file_path
-        self._service_timeout = service_timeout
+    def __init__(self, config_file_path_, service_timeout_, excluded_joints_=None):
```
Comment (ethanfowler) [inline sr_utilities/scripts/sr_utilities/controller_spawner.py]: I think the trailing `_` suffix is only recommended for avoiding conflicts with python keywords, not variable names that we have full control of!

**Review by ethanfowler — CHANGES_REQUESTED** [review]

File: sr_utilities/scripts/interactive_marker.py (hunk 1 above)
Comment (carebare47) [reply sr_utilities/scripts/interactive_marker.py]: 55a6f2f7c523c3c3b14b1771ed976b52db5199b5

File: sr_utilities/scripts/interactive_marker.py (hunk 3 above)
Comment (carebare47) [reply sr_utilities/scripts/interactive_marker.py]: main removed here 55a6f2f7c523c3c3b14b1771ed976b52db5199b5

File: sr_utilities/scripts/interactive_marker.py (hunk 1 above)
Comment (carebare47) [reply sr_utilities/scripts/interactive_marker.py]: `.keys()` still triggers `C0206`, but I've made the change here anyway (agree that it's clearer to use `.keys()` ) 7113a011792b2cdba54e7bb8845356298efb715b

File: sr_hand/scripts/sr_hand/shadowhand_commander.py (hunk 5 above)
Comment (carebare47) [reply sr_hand/scripts/sr_hand/shadowhand_commander.py:26]: Docstrings have got to be inside classes, moved it here 7113a011792b2cdba54e7bb8845356298efb715b

File: sr_hand/scripts/sr_hand/shadowhand_commander.py (hunk 6 above)
Comment (carebare47) [reply sr_hand/scripts/sr_hand/shadowhand_commander.py]: 7113a011792b2cdba54e7bb8845356298efb715b

File: sr_hand/scripts/sr_hand/shadowhand_commander.py (hunk 7 above)
Comment (carebare47) [reply sr_hand/scripts/sr_hand/shadowhand_commander.py]: yeah that's better, thanks 7113a011792b2cdba54e7bb8845356298efb715b

File: sr_hand/scripts/sr_hand/shadowhand_commander.py (hunk 8 above)
Comment (carebare47) [reply sr_hand/scripts/sr_hand/shadowhand_commander.py]: 7113a011792b2cdba54e7bb8845356298efb715b

File: sr_hand/scripts/sr_hand/shadowhand_commander.py (hunk 9 above)
Comment (carebare47) [reply sr_hand/scripts/sr_hand/shadowhand_commander.py]: 7113a011792b2cdba54e7bb8845356298efb715b

File: sr_tactile_sensors/scripts/sr_tactile_sensors/virtual_ShadowPST_publisher.py (hunk 10 above)
Comment (carebare47) [reply sr_tactile_sensors/scripts/sr_tactile_sensors/virtual_ShadowPST_publisher.py]: 7113a011792b2cdba54e7bb8845356298efb715b

File: sr_utilities/scripts/sr_utilities/controller_spawner.py (hunk 11 above)
Comment (carebare47) [reply sr_utilities/scripts/sr_utilities/controller_spawner.py]: good point about trailing `_` being for python keywords. I've added leading underscores in the main instead, thoughts? 7113a011792b2cdba54e7bb8845356298efb715b

File: sr_utilities/scripts/interactive_marker.py (hunk 3 above)
Comment (carebare47) [reply sr_utilities/scripts/interactive_marker.py]: removed from install python executables here b003c18c2a69cd84f105ddd699ab596fda1574c9 (and fixed typo here d89926e08d94faa8a694bf824475c3865a4da66e)

File: sr_utilities/scripts/sr_utilities/controller_spawner.py (hunk 11 above)
Comment (ethanfowler) [reply sr_utilities/scripts/sr_utilities/controller_spawner.py]: I don't usually prepend `__main__`-scope variables with `_`, just class member variables that I want to indicate are "protected". Using it in `__main__` doesn't do any harm, but might confuse people?

Comment (ethanfowler) [reply sr_utilities/scripts/sr_utilities/controller_spawner.py]: https://google.github.io/styleguide/pyguide.html#3162-naming-conventions

Comment (carebare47) [reply sr_utilities/scripts/sr_utilities/controller_spawner.py]: 922a19db3bc6e43f71f1e48d736f2eae0b2d9ba5

Comment (ethanfowler) [reply sr_utilities/scripts/sr_utilities/controller_spawner.py]: Looks good to me

**Review by ethanfowler — APPROVED** [review]

**Review by beatrizleon — APPROVED** [review]

**Pull request merged**